src/base/ApplicationLoadBalancer.ts

```diff
@@ -64,12 +84,95 @@ export class ApplicationLoadBalancer extends Resource {
       },
     });
 
-    this.alb = new elb.Alb(this, `alb`, {
+    let logsConfig: elb.AlbAccessLogs = undefined;
+    if (config.accessLogs !== undefined) {
+      //If logs were configured lets set them up
+      const accountId = new datasources.DataAwsCallerIdentity(this, 'caller')
+        .accountId;
+
+      const accessLogsPrefix = `server-logs/${config.prefix.toLowerCase()}/internal-alb/AWSLogs/${accountId}/elasticloadbalancing/`;
+      const prefix =
+        config.accessLogs.prefix !== undefined ? accessLogsPrefix : '';
+
+      const bucket = this.getOrCreateBucket({
+        bucket: config.accessLogs.bucket,
+        existingBucket: config.accessLogs.existingBucket,
+        prefix,
+        accountId,
+      });
+
+      logsConfig = {
+        bucket,
+        enabled: true,
+        prefix,
+      };
+    }
+
+    const albConfig: elb.AlbConfig = {
       namePrefix: config.alb6CharacterPrefix,
       securityGroups: [this.securityGroup.id],
       internal: config.internal !== undefined ? config.internal : false,
       subnets: config.subnetIds,
       tags: config.tags,
+      accessLogs: logsConfig,
+    };
+    this.alb = new elb.Alb(this, `alb`, albConfig);
+  }
+
+  /**
+   *
+   * @param config Creates a bucket according to https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html#attach-bucket-policy if one does not exist
+   * @returns
+   */
+  private getOrCreateBucket(config: {
+    existingBucket?: string;
+    bucket?: string;
+    accountId: string;
+    prefix: string;
+  }): string {
+    if (config.existingBucket === undefined && config.bucket === undefined) {
+      throw new Error(
+        'If you are configuring access logs you need to define either an existing bucket or a new one to store the logs'
+      );
+    }
```

Comment:
do you think we should also throw an error if `prefix` doesn't have a trailing slash?